x-pack/plugins/observability_solution/apm/public/components/routing/templates/apm_main_template/index.tsx

```diff
@@ -158,13 +163,17 @@ export function ApmMainTemplate({
 
   const showEntitiesInventoryCallout =
     !dismissedEntitiesInventoryCallout &&
-    isEntityCentricExperienceSettingEnabled &&
+    isEntityCentricExperienceEnabled &&
     selectedNavButton !== undefined;
 
   return (
     <EnvironmentsContextProvider>
       <ObservabilityPageTemplate
-        noDataConfig={shouldBypassNoDataScreen ? undefined : noDataConfig}
+        noDataConfig={
+          shouldBypassNoDataScreen || (isEntityExperienceFullyEnabled && hasLogsData)
```

Comment:
This makes the logic a bit confusing, we have well defined variable `shouldBypassNoDataScreen` but that is not the source of truth to show or not the no data screen.